ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -184,6 +186,16 @@ def session(self) -> Session:
 
         return self._session
 
+    @property
+    def connection(self) -> Connection:
```

Comment:
we only use this connection for generate_sample_data we should move this into that method and only get it if the generate_sample_data is set to True

Reply:
I use this connection for the profiler as well to not create more than one. Felt a good call to have it specific to the source. Shall we move that?

Reply:
ok actually I use the engine not the connection, got confused. I'll move this. Thanks

Reply:
🤔

Reply:
the connection will get initialised only on call. If sampleData is false, nothing will run here. Thanks 🙏